src/pages/billing/Project.js

```diff
@@ -289,26 +287,21 @@ const ProjectDetail = ({ project, billingAccounts, authorizeAndLoadAccounts }) =
   })
 
   // Helpers
-  const updateBillingProject = withErrorReporting('Error updating billing project')(
-    () => Ajax(signal).Billing.billingProject(billingProject.projectName).then(setBillingProject)
+  const setBillingAccount = withErrorReporting('Error updating billing account')(
+    async newAccountName => {
+      Ajax().Metrics.captureEvent(Events.changeBillingAccount, {
+        oldName: billingProject.billingAccount,
+        newName: newAccountName,
+        billingProjectName: billingProject.projectName
+      })
+      await Ajax(signal).Billing.changeBillingAccount({
```

Comment:
TOL: If you `return` here instead of `await`ing, you wouldn't have to make the function `async`; I don't know which I prefer, though, or if I have a preference.

Reply:
I think I prefer this - I returned the promise and then chained off it in the onClick handler to `setBillingAccount(selectedBilling).then(updateProjectAndReload)`

Reply:
To be clear, there's no functional difference between returning this promise and awaiting it, as creating an async function is just a promise itself.